java/src/main/java/gov/doc/ntia/sigmf/ext/global/algorithm/TraceDataProduct.java

```diff
@@ -0,0 +1,61 @@
+package gov.doc.ntia.sigmf.ext.global.algorithm;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import java.util.ArrayList;
+import java.util.List;
+
+public abstract class TraceDataProduct extends DataProduct {
+  public List<Trace> traces;
+
+  @JsonProperty(required = false)
+  protected String reference;
+
+  @JsonProperty(value = "units", required = false)
+  protected String units;
+
+  @JsonProperty(required = true)
+  protected Long samples;
+
+  public void addTrace(Trace trace) {
+    if (traces == null) {
+      traces = new ArrayList<>();
+    }
+    traces.add(trace);
+  }
+
+  public void removeDetector(int i) {
```

Comment:
naming inconsistent between `addTrace()` and `removeDetector()`. Suggest to rename `removeDetector()` to `removeTrace()`